File: example/src/main/java/com/github/simonpercic/oklogexample/AppComponent.java
package com.github.simonpercic.oklogexample;

import com.github.simonpercic.oklogexample.data.api.ApiService;

import javax.inject.Singleton;

import dagger.Component;

/**
 * App's Dagger Component.
 *
 * @author Simon Percic <a href="https://github.com/simonpercic">https://github.com/simonpercic</a>
 */
@Singleton
@Component(
        modules = {
                AppModule.class,
                FlavorAppModule.class
        }
)
public interface AppComponent {

    void inject(App app);

    ApiService apiService();
}

File: example/src/main/kotlin/com/github/simonpercic/oklogexample/AppComponent.kt
package com.github.simonpercic.oklogexample

import com.github.simonpercic.oklogexample.data.api.ApiService
import dagger.Component
import javax.inject.Singleton

/**
 * App's Dagger Component.

 * @author Simon Percic [https://github.com/simonpercic](https://github.com/simonpercic)
 */
@Singleton
@Component(modules = arrayOf(AppModule::class, FlavorAppModule::class))
interface AppComponent {

    fun inject(app: App)

    fun apiService(): ApiService
}
